### PR TITLE
feat(dual-list-selector): add support for disabled items

### PR DESCRIPTION
**Pull request by KKoukiou**
Needed for: https://github.com/patternfly/patternfly-react/issues/6250

### CODE REVIEW

File: src/patternfly/components/DualListSelector/dual-list-selector-item.hbs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<{{#if dual-list-selector-item--type}}{{dual-list-selector-item--type}}{{else}}button{{/if}} class="pf-c-dual-list-selector__item{{#if dual-list-selector-item--HasCheck}} pf-m-check{{/if}}{{#if dual-list-selector-item--IsSelected}} pf-m-selected{{/if}}{{#if dual-list-selector-item--modifier}} {{dual-list-selector-item--modifier}}{{/if}}"
+<{{#if dual-list-selector-item--type}}{{dual-list-selector-item--type}}{{else}}button{{/if}} class="pf-c-dual-list-selector__item{{#if dual-list-selector-item--HasCheck}} pf-m-check{{/if}}{{#if dual-list-selector-item--IsSelected}} pf-m-selected{{/if}}{{#if dual-list-selector-item--IsDisabled}} pf-m-disabled{{/if}}{{#if dual-list-selector-item--modifier}} {{dual-list-selector-item--modifier}}{{/if}}"
```
Comment (mcoker) [inline src/patternfly/components/DualListSelector/dual-list-selector-item.hbs:1]: When the item is a `<button>`, you should only need to add a `disabled` attribute, then all you need to do is style the text via `:disabled`.

But this item can also be a `<div>` for both expandable and checkbox items. Maybe in those cases, it's appropriate to add `.pf-m-disabled` here, remove the `tabindex` from this element, add `disabled` to the check,  and style the toggle and text via `.pf-m-disabled` as you've done here.

File: src/patternfly/components/DualListSelector/dual-list-selector.scss [hunk 2]
```diff
@@ -219,6 +220,15 @@ $pf-c-dual-list-selector__item--MaxNesting: 10;
     }
   }
 
+  &.pf-m-disabled {
+    .pf-c-dual-list-selector__item-text {
+      --pf-c-dual-list-selector__item-text--Color: var(--pf-c-dual-list-selector__item--m-disabled__text--Color);
+    }
+
+    pointer-events: none;
+    cursor: none;
```
Comment (mcoker) [inline src/patternfly/components/DualListSelector/dual-list-selector.scss:229]: If you add `pointer-events: none;`, you shouldn't need to style the `cursor`

**Review by mcoker — CHANGES_REQUESTED** [review]
Thanks for the PR @KKoukiou!!! Left some comments. Also added @mcarrano to review visually.

**Comment by KKoukiou** [comment]
> This looks fine to me. However I'm not quite sure how this addresses [patternfly/patternfly-react#6250](https://github.com/patternfly/patternfly-react/issues/6250) which seems to be asking to disable the entire control. Would the idea be that `isDisabled` would just set all items to disabled so you couldn't select anything? I guess this works, but not sure when that's useful.

It's much more flexible to be able to disable each of the items, I can see such a scenario for cockpit for example, where some of the items couldn't be used. If you want to disable all you can still do that by disabling each of them separately.

**Review by mcarrano — APPROVED** [review]
LGTM!

**Comment by mcoker** [comment]
@KKoukiou do you need to get this in for this release? If so we have a code freeze today. If we can update the styles/examples to use `:disabled` and remove the `cursor` styles, we can merge it and follow up on disabled expandable/check items.

**Comment by KKoukiou** [comment]
@mcoker no it's not urgent from my side.

File: src/patternfly/components/DualListSelector/dual-list-selector.scss (hunk 2 above)
Comment (KKoukiou) [reply src/patternfly/components/DualListSelector/dual-list-selector.scss:229]: I copied this from src/patternfly/components/DataList/data-list.scss - we can fix this there as well then.

**Comment by KKoukiou** [comment]
@mcoker I am not sure if I did the needed changes, let me know.

File: src/patternfly/components/DualListSelector/dual-list-selector.scss [hunk 3]
```diff
@@ -259,6 +260,14 @@ $pf-c-dual-list-selector__item--MaxNesting: 10;
   text-align: left;
   cursor: pointer;
   border: 0;
+
+  &.pf-m-disabled {
+    .pf-c-dual-list-selector__item-text {
+      --pf-c-dual-list-selector__item-text--Color: var(--pf-c-dual-list-selector__list-item-row--m-disabled__text--Color);
```
Comment (mcoker) [inline src/patternfly/components/DualListSelector/dual-list-selector.scss:266]: You can just set this var under `&.pf-m-disabled` - should be no need to scope it to `&.pf-m-disabled .pf-c-dual-list-selector__item-text`

File: src/patternfly/components/DualListSelector/dual-list-selector-item.hbs [hunk 4]
```diff
@@ -1,7 +1,12 @@
 {{#> dual-list-selector-list-item-row}}
-  <{{#if dual-list-selector-item--type}}{{dual-list-selector-item--type}}{{else}}button{{/if}} class="pf-c-dual-list-selector__item{{#if dual-list-selector-item--modifier}} {{dual-list-selector-item--modifier}}{{/if}}"
-    {{#if dual-list-selector-item--type}}
-      tabindex="0"
+  <{{#if dual-list-selector-item--type}}{{dual-list-selector-item--type}}{{else}}button{{/if}} class="pf-c-dual-list-selector__item{{#if dual-list-selector-item--IsDisabled}} pf-m-disabled{{/if}}{{#if dual-list-selector-item--modifier}} {{dual-list-selector-item--modifier}}{{/if}}"
```
Comment (mcoker) [inline src/patternfly/components/DualListSelector/dual-list-selector-item.hbs:2]: For this block I think you want to add some checks around `pf-m-disabled` and the `disabled` attr. A button should have `disabled` and a div/non-button should have `pf-m-disabled`.

```
<{{#if dual-list-selector-item--type}}{{dual-list-selector-item--type}}{{else}}button{{/if}} class="pf-c-dual-list-selector__item
  {{~#if dual-list-selector-item--IsDisabled}}
    {{~#if dual-list-selector-item--type}} pf-m-disabled{{/if}} // add pf-m-disabled if it's a disabled div/non-button
  {{/if}}
  {{~#if dual-list-selector-item--modifier}} {{dual-list-selector-item--modifier}}{{/if}}"
  {{#unless dual-list-selector-item--IsDisabled}}
    {{#if dual-list-selector-item--type}}
      tabindex="0"
    {{/if}}
  {{/unless}}
  {{#if dual-list-selector-item--IsDisabled}}
    {{#unless dual-list-selector-item--type}}
      disabled // add disabled attr if it's a disabled button
```

File: src/patternfly/components/DualListSelector/dual-list-selector.scss [hunk 5]
```diff
@@ -259,6 +260,14 @@ $pf-c-dual-list-selector__item--MaxNesting: 10;
   text-align: left;
   cursor: pointer;
   border: 0;
+
+  &.pf-m-disabled {
```
Comment (mcoker) [inline src/patternfly/components/DualListSelector/dual-list-selector.scss:264]: add `:disabled` to this:

```
&.pf-m-disabled, // non-buttons
&:disabled { // buttons
```

**Review by mcoker — CHANGES_REQUESTED** [review]
Thanks for the updates @KKoukiou!! Left some feedback. Looks like you're close to supporting both buttons and expandable items (`div`s). Would you mind adding an example of a disabled expandable item? We will also need to disable all nested items of an expandable item.

File: src/patternfly/components/DualListSelector/dual-list-selector.scss [hunk 6]
```diff
@@ -53,6 +53,7 @@ $pf-c-dual-list-selector__item--MaxNesting: 10;
   --pf-c-dual-list-selector__item-text--Color: var(--pf-global--Color--100);
   --pf-c-dual-list-selector__list-item-row--m-selected__text--Color: var(--pf-global--active-color--100);
   --pf-c-dual-list-selector__list-item-row--m-selected__text--FontWeight: var(--pf-global--FontWeight--bold);
+  --pf-c-dual-list-selector__list-item-row--m-disabled__text--Color: var(--pf-global--disabled-color--200);
```
[the remark this answers is not in the document]
Comment (mcarrano) [reply src/patternfly/components/DualListSelector/dual-list-selector.scss:56]: @mdoruskova @mceledonia can you confirm what the correct color for disabled menu items should be?

**Comment by KKoukiou** [comment]
@mcoker thanks a lot for the hints! Updated according to your comments. Sorry for the initial mess - I am not very much in touch with this repo.

**Review by mcoker — CHANGES_REQUESTED** [review]
Just realized we will need a disabled class at least on the list-item-row element since that's where we change the background color of the item on hover. I wanted to run through the changes to make sure it will work so I issued a PR to your branch here if you'd like to see what I did - https://github.com/KKoukiou/patternfly/pull/1

It also:
* disables the check for disabled items
* applies a disabled color to an expandable toggle icon
* limits the `disabled` attribute to items that are buttons
* moves `pf-m-disabled` to a top level `__list-item` element

**Comment by KKoukiou** [comment]
> Just realized we will need a disabled class at least on the list-item-row element since that's where we change the background color of the item on hover. I wanted to run through the changes to make sure it will work so I issued a PR to your branch here if you'd like to see what I did - [KKoukiou#1](https://github.com/KKoukiou/patternfly/pull/1)
> 
> It also:
> 
>     * disables the check for disabled items
> 
>     * applies a disabled color to an expandable toggle icon
> 
>     * limits the `disabled` attribute to items that are buttons
> 
>     * moves `pf-m-disabled` to a top level `__list-item` element

Ah cool :) As I am really strugling with CSS I feel I am abusing your review to learn here. thanks for fixing it for me!

**Review by mcoker — APPROVED** [review]

**Review by srambach — APPROVED** [review]
LGTM, thanks for the docs update.

**Pull request merged**